src/rgw/rgw_rados.cc

```diff
@@ -9073,22 +9073,45 @@ int RGWRados::remove_objs_from_index(RGWBucketInfo& bucket_info, list<rgw_obj_in
   if (r < 0)
     return r;
 
-  bufferlist updates;
+  unsigned int skip = 0;
```

Comment:
is there no way to use range-for or iterator looping here?  do loops with multiple exits have been a source of bugs

Reply:
@mattbenjamin I have changed it with range-for